internal/pkg/cli/project_init.go

```diff
@@ -4,60 +4,166 @@
 type InitProjectOpts struct {
 	ProjectName string
-	manager     archer.ProjectCreator
-	ws          archer.Workspace
+
+	projectStore archer.ProjectStore
+	ws           archer.Workspace
+	prompt       prompter
+}
+
+// NewInitProjectOpts returns a new InitProjectOpts.
+func NewInitProjectOpts() (*InitProjectOpts, error) {
+	ssmStore, err := ssm.NewStore()
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+	return &InitProjectOpts{
+		projectStore: ssmStore,
+		ws:           ws,
+		prompt:       prompt.New(),
+	}, nil
+}
+
+// Ask prompts the user for any required arguments that they didn't provide.
+func (opts *InitProjectOpts) Ask() error {
+	if opts.ProjectName != "" {
+		// User passed in a name as an argument.
+		return nil
+	}
+
+	// If there's a local project, we'll use that and just skip the project question.
+	// Otherwise, we'll load a list of existing projects that the customer can select from.
+	if summary, err := opts.ws.Summary(); err == nil {
+		log.Infof("Looks like you are using a workspace that's registered to project %s. We'll use that as your project.\n", color.HighlightUserInput(summary.ProjectName))
+		opts.ProjectName = summary.ProjectName
+		return nil
+	}
+
+	existingProjects, _ := opts.projectStore.ListProjects()
+	if len(existingProjects) == 0 {
+		log.Infoln("Looks like you don't have any existing projects. Let's create one!")
+		return opts.askNewProjectName()
+	}
+
+	log.Infoln("Looks like you have some projects already.")
+	useExistingProject, err := opts.prompt.Confirm("Would you like to use one of your existing projects?", "", prompt.WithTrueDefault())
+	if err != nil {
+		return fmt.Errorf("new project confirmation: %w", err)
+	}
+	if useExistingProject {
+		log.Infoln("Ok, here are your existing projects.")
+		return opts.askSelectExistingProjectName(existingProjects)
+	}
+	log.Infoln("Ok, let's create a new project then.")
+	return opts.askNewProjectName()
+}
+
+// Validate returns an error if the user's input is invalid.
+func (opts *InitProjectOpts) Validate() error {
+	return validateProjectName(opts.ProjectName)
 }
 
 // Execute creates a new managed empty project.
 func (opts *InitProjectOpts) Execute() error {
-	if err := validateProjectName(opts.ProjectName); err != nil {
-		return err
-	}
-	if err := opts.manager.CreateProject(&archer.Project{
+	err := opts.projectStore.CreateProject(&archer.Project{
 		Name: opts.ProjectName,
-	}); err != nil {
-		return err
-	}
+	})
 
+	if err != nil {
+		// If the project already exists, move on - otherwise return the error.
+		var projectAlreadyExistsError *store.ErrProjectAlreadyExists
+		if !errors.As(err, &projectAlreadyExistsError) {
+			return err
+		}
+	}
 	return opts.ws.Create(opts.ProjectName)
 }
 
+func (opts *InitProjectOpts) RecommendedActions() []string {
+	return []string{
+		fmt.Sprintf("Run %s to add a new application to your project.", color.HighlightCode("archer init")),
```

Comment:
Will all of these `color.HighlightCode` break on Windows?

Reply:
Jes